keras_cv/layers/preprocessing/random_hue.py

```diff
@@ -0,0 +1,66 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import tensorflow as tf
+
+from keras_cv.utils import preprocessing
+
+
+class RandomHue(tf.keras.__internal__.layers.BaseImageAugmentationLayer):
+    """Randomly adjusts the hue on given images.
+
+    This layer will randomly increase/reduce the hue for the input RGB
+    images. At inference time, the output will be identical to the input.
+    Call the layer with `training=True` to adjust the brightness of the input.
+
+    The image hue is adjusted by converting the image(s) to HSV and rotating the
+    hue channel (H) by delta. The image is then converted back to RGB.
+
+    Args:
+        factor: Either a tuple of two floats or a single float. `factor` controls the
+            extent to which the image saturation is impacted. `factor` =
+            `0.0`, `0.5` or `1.0` makes this layer perform a no-op operation.
+            `factor=0.25` and `factor=0.75` makes the image to have fully opposite
+            hue value. Values should be between `0.0` and `1.0`.
+            If a tuple is used, a `factor` is sampled
+            between the two values for every image augmented.  If a single float is
+            used, a value between `0.0` and the passed float is sampled.
+            In order to ensure the value is always the same, please pass a tuple with
+            two identical floats: `(0.5, 0.5)`.
+    """
+
+    def __init__(self, factor, **kwargs):
```

Comment:
Can we actually provide a sensible default for this layer's factor?  Mainly because the range is a little confusing

Reply:
It really depends on how user want to adjust the color. If we have to come up a default value, I would just give [0, 1]